app/src/main/java/com/example/namp5/tipcalculator/MainActivity.java

```diff
@@ -1,13 +1,249 @@
+                    mOperator = NUMBER_SUB;
+                } else if (mSaveResult != NUMBER_0) {
+                    mResult = mSaveResult;
+                    mTextResult.setText("");
+                    mTextResult.setText((parseFloatToInt(mResult) + NUMBER_SUB ));
+                    mTextInputNumber.setText("");
+                    mOperator = NUMBER_SUB;
+                }
+                break;
+            case R.id.text_multiplication:
+                if (mSaveResult == 0) {
+                    mResult = Float.parseFloat(mTextInputNumber.getText().toString());
+                    mTextInputNumber.append(NUMBER_MUL);
+                    mTextResult.setText(mTextInputNumber.getText().toString());
+                    mTextInputNumber.setText("");
+                    mOperator = NUMBER_MUL;
+                } else if (mSaveResult != NUMBER_0) {
+                    mResult = mSaveResult;
+                    mTextResult.setText("");
+                    mTextResult.setText((parseFloatToInt(mResult) + NUMBER_MUL ));
+                    mTextInputNumber.setText("");
+                    mOperator = NUMBER_MUL;
+                }
+                break;
+            case R.id.text_division:
+                if (mSaveResult == 0) {
+                mResult = Float.parseFloat(mTextInputNumber.getText().toString());
+                mTextInputNumber.append(NUMBER_DIV);
+                mTextResult.setText(mTextInputNumber.getText().toString());
+                mTextInputNumber.setText("");
+                mOperator = NUMBER_DIV;
+            } else if (mSaveResult != NUMBER_0) {
+                mResult = mSaveResult;
+                mTextResult.setText("");
+                mTextResult.setText((parseFloatToInt(mResult) + NUMBER_DIV));
+                mTextInputNumber.setText("");
+                mOperator = NUMBER_DIV;
+            }
+                break;
+            case R.id.text_division_100:
+                if(mSaveResult !=NUMBER_0){
+                    mResult = Float.parseFloat(mTextInputNumber.getText().toString());
+                    mTextInputNumber.append(NUMBER_DIV_100);
+                    mTextResult.setText(mTextInputNumber.getText().toString());
+                    mTextInputNumber.setText("");
+                    mOperator = NUMBER_DIV_100;
+                }
+                break;
+            case R.id.text_plus_subtraction:
+                if(mTextInputNumber.getText().toString().contains("-")){
+                    mTextInputNumber.setText(mTextInputNumber.getText().toString().substring(1));
+                    mSaveResult = Float.parseFloat(mTextInputNumber.getText().toString().substring(1));
+                }else{
+                    mTextInputNumber.setText(NUMBER_SUB + mTextInputNumber.getText().toString());
+                    mSaveResult = Float.parseFloat(mTextInputNumber.getText().toString());
+                }
+                break;
+            case R.id.text_equals:
```

Comment:
case này sao bên trong to thế này

Reply:
Em tách code trong case này ra thành phương thức riêng rồi ạ nhưng nó không báo outdated